PolyEngine/Engine/Res/Shaders/blinn-phongFrag.shader

```diff
@@ -45,6 +54,8 @@ uniform DirectionalLight uDirectionalLight[8];
 uniform int uDirectionalLightCount;
 uniform PointLight uPointLight[8];
 uniform int uPointLightCount;
+uniform SpotLight uSpotLight[8];
```

Comment:
Don't use magic numbers, extract this to define on the top of the file. Like so:
```
#ifndef MAX_SPOTLIGHT_COUNT
#define MAX_SPOTLIGHT_COUNT 8
#endif
```
This way, in the future, we can provide variable count of lights during shader compilation, by appending `#define MAX_SPOTLIGHT_COUNT 12` to the beginning of the shader file. Do it also for the point lights and directional lights. This should also be applied to every constant in shader code.